File: app/src/main/java/kandroid/data/ArsenalData.java
package kandroid.data;

import java.util.Date;

import kandroid.observer.POJO;
import kandroid.utils.Identifiable;

public class ArsenalData extends Data<ArsenalData.ApiKdock> implements Identifiable {

    public int getArsenalID() {
        return data.api_id;
    }

    public int getState() {
        return data.api_state;
    }

    public int getShipID() {
        return data.api_created_ship_id;
    }

    public Date getCompletionTime() {
        return new Date(data.api_complete_time);
    }

    public int getFuel() {
        return data.api_item1;
    }

    public int getAmmo() {
        return data.api_item2;
    }

    public int getSteel() {
        return data.api_item3;
    }

    public int getBauxite() {
        return data.api_item4;
    }

    public int getDevelopmentMaterial() {
        return data.api_item5;
    }

    @Override
    public int getID() {
        return data.api_id;
    }

    public static class ApiKdock extends POJO {
        public int api_id;
        public int api_state;
        public int api_created_ship_id;
        public int api_complete_time;
        public String api_complete_time_str;
        public int api_item1;
        public int api_item2;
        public int api_item3;
        public int api_item4;
        public int api_item5;
    }
}

File: app/src/main/java/kandroid/data/ArsenalData.kt
package kandroid.data

import kandroid.observer.kcsapi.api_req_kousyou
import kandroid.utils.Identifiable
import kandroid.utils.json.get
import kandroid.utils.json.int
import kandroid.utils.json.long
import kandroid.utils.json.set
import java.util.*

class ArsenalData : JsonWrapper(), RequestDataListener, Identifiable {

    val arsenalId: Int get() = data["api_id"].int()

    val state: Int get() = data["api_state"].int()

    val shipID: Int get() = data["api_created_ship_id"].int()

    val completionTime: Date get() = Date(data["api_complete_time"].long())

    val fuel: Int get() = data["api_item1"].int()

    val ammo: Int get() = data["api_item2"].int()

    val steel: Int get() = data["api_item3"].int()

    val bauxite: Int get() = data["api_item4"].int()

    val developmentMaterial: Int get() = data["api_item5"].int()

    override val id: Int get() = arsenalId

    override fun loadFromRequest(apiName: String, requestData: MutableMap<String, String>) {
        when (apiName) {
            api_req_kousyou.createship_speedchange.name -> {
                data["api_state"] = 3
            }
        }
    }
}
